site/public/js/server.js

```diff
@@ -29,6 +29,14 @@ function adjustBreadcrumbLinks(){
 }
 ////////////End: Removed redundant link in breadcrumbs//////////////////////////
 
+/* GLOBAL VARIABLES */
+
+// Map to store the XMLHttpRequest object returned by getJSON when getting the 
+// output to display in loadTestcaseOutput()
+// Key: orig_div_name   Value: XMLHttpRequest object
+var output_xhrs = new Map();
```

Comment:
Why do we need a global variable here?  Global variables are generally a very very very bad thing in terms of future development.

Reply:
The global variable and functions using it have been moved to their own JavaScript file.